### PR TITLE
Document advantage of using arrays for light travel-time calculations

### LINKED ISSUE #7755: DOC: avoid iteration for light travel time of lots of objects
In our user survey, @stargaser mentioned they had issues at ZTF when calculating light travel times for tens of thousands of sources in a degree patch of the sky. It might be worth a section in the performance area of the ```Time``` documentation comparing run times of the following approaches:

```python
import numpy as np
import astropy.coordinates as coord
import astropy.units as u
from astropy.time import Time

ra = np.random.normal(0.0, 1.0, 50000)
dec = np.random.normal(0.0, 1.0, 50000)

coos = coord.SkyCoord(ra, dec, unit=u.deg)
observatory = coord.EarthLocation.of_site('lapalma')

%time time.light_travel_time(coos, location=observatory)
CPU times: user 56 ms, sys: 3.21 ms, total: 59.2 ms
Wall time: 58.2 ms
<TimeDelta object: scale='tdb' format='jd' value=[ 0.00497521  0.00495056  0.00497048 ...,  0.00499594  0.0049838
  0.00502038]>

%time ltts = [time.light_travel_time(coo, location=coos.location) for coo in coos]
CPU times: user 16min 45s, sys: 5.08 s, total: 16min 50s
Wall time: 16min 58s
```

### PR DESCRIPTION
**Pull request by SG004**
New Attempt at Issue #7755, related to the now closed PR #7893 

This PR was started in October of last year. I deleted my fork and re-forked after the new year, so the old branch is no longer. If there are any suggestions, please let me know so I can address them as fast as I can. 

EDIT: Fix #7755

### CODE REVIEW

**Comment by mhvk** [comment]
Didn't look at the PR beyond trying to understand what it was about - and hence gave it a more meaningful title.

**Comment by SG004** [comment]
@bsipocz  @pllim So the iteration over the 50,000 points causes a memory error. Should I lower the number in order to prevent the memory error from taking place?

**Comment by pllim** [comment]
How about 10k? @eteq or @adrn have the final say, though.

**Comment by bsipocz** [comment]
fyi: to get the site info for La Palma, you need to have internet connection and that is switched off for most of the test runs.

**Comment by SG004** [comment]
@bsipocz That is good to know

**Comment by SG004** [comment]
@pllim @eteq @adrn 
The link below is a gist of the Colab I used when timing the 10000 points. Let me know if I should use 10k and update the PR with the new value and associated times. 
https://gist.github.com/SG004/5ff3e4c6e258658299b9caf4629bd34a

**Comment by SG004** [comment]
@pllim  @eteq  @bsipocz Now that I've fixed the syntax mistakes and other various issues. The issue and the reason for the failing seems to be that it is taking over 10 minutes in the 32bit as well as something to do with
astropy.coordinates.name_resolve.NameResolveError: Unable to find coordinates for name 'UMa II' - in the travis CI

P.S. I will squash all the commits once everything passes

**Comment by pllim** [comment]
Hmm... I am starting to think either we should skip that doctest or mark it as "slow". But do we have the capability of running "slow doctest" only when certain option is passed in?

**Comment by bsipocz** [comment]
> I am starting to think either we should skip that doctest or mark it as "slow".

@pllim - do we have  the infrastructure to do this? I suppose we can easily use the pytest slow decorator, but how to do that in doctests?

**Comment by pllim** [comment]
> but how to do that in doctests?

@drdavella ?

**Comment by drdavella** [comment]
It can't be done right now. Feel free to submit a PR to [pytest-doctestplus](https://github.com/astropy/pytest-doctestplus/pull/41).

**Comment by astrofrog** [comment]
Removing the milestone since this is still pending, but since it's just a doc fix it can go into a bugfix release if needed

**Comment by pllim** [comment]
:+1: to the doc but I am still hesitant to let a "slow example" run for every push event.

File: docs/time/performance.inc.rst [hunk 1]
```diff
@@ -10,3 +10,76 @@
...
+   **CPU times:** user 31.4 ms, sys: 1 ms, total: 32.4 ms
+
+   **Wall time:** 31.8 ms
+
+Approach Using Iteration
+^^^^^^^^^^^^^^^^^^^^^^^^
+Iterating over a SkyCoord object is not fast, due to the how a SkyCoord is constructed. Therefore
+the user time is significantly longer.
+
+   >>> ltts = [Time.light_travel_time(self=Time.now(),skycoord=coos, location=observatory) for coo in coos]
```
Comment (ayshih) [inline docs/time/performance.inc.rst]:
```suggestion
   >>> ltts = [Time.light_travel_time(self=Time.now(), skycoord=coo, location=observatory) for coo in coos]
```
It looks like the code is mistakenly converting all of the coordinates with each iteration.  The time measurements should be redone.

Comment (astrofrog) [reply docs/time/performance.inc.rst]: It would probably make sense to avoid doing ``Time.now()`` for each item?

Comment (YasPHP) [reply docs/time/performance.inc.rst:57]: ^ right, you'd probably want to calculate all the times in a single batch and access each relevant one with each of the iterations. That redundant time calculation is probably what's slowing this down massively.

**Pull request closed without merging**